web-module/src/main/java/by/tut/mdcatalog/project2/web/controller/ArticleController.java

```diff
@@ -2,39 +2,95 @@
     private final ArticleService articleService;
     private final CommentService commentService;
-    public ArticleController(ArticleService articleService, CommentService commentService) {
+    private final UserService userService;
+
+
+    public ArticleController(ArticleService articleService,
+                             CommentService commentService,
+                             UserService userService) {
         this.articleService = articleService;
         this.commentService = commentService;
+        this.userService = userService;
     }
 
     @GetMapping("/articles")
     public String getArticles(Model model) {
-        List<ArticleDTO> articlesDTO = articleService.getArticles();
+        List<ArticleDTO> articlesDTO = articleService.getAll();
         model.addAttribute("articles", articlesDTO);
         return "articles";
     }
 
     @GetMapping("/articles/{id}")
-    public String getArticle (@PathVariable(value = "id") Long id, ModelMap modelMap) {
+    public String getArticle(@PathVariable(value = "id") Long id, ModelMap modelMap) {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String username = authentication.getName();
+        UserDTO currentUserDTO = userService.getByUsername(username);
+
         ArticleDTO articleDTO = articleService.getById(id);
         List<CommentDTO> commentsDTO = commentService.getByArticleId(id);
         modelMap.addAttribute("article", articleDTO);
         modelMap.addAttribute("comments", commentsDTO);
+        modelMap.addAttribute("currentuserid", currentUserDTO.getId());
         return "article";
     }
+
+    @PostMapping("/articles/{id}")
+    public String changeArticleInfo(@PathVariable(name = "id") Long id,
+                                    @Valid ArticleDTO articleDTO,
+                                    BindingResult result,
+                                    Model model) {
+        if (result.hasErrors()) {
+            model.addAttribute("article", articleDTO);
+            logger.info("Article update invalid (ID) {}", id);
+            return "article";
+        }
+        articleDTO.setId(id);
+        articleDTO.setDate(new Date());
+        articleService.update(articleDTO);
+        logger.info("Article updated (ID) {}", id);
+        return "redirect:/success";
+    }
+
+    @PostMapping("/articles/comments")
+    public String deleteComments(@RequestParam(value = "ids", required = false) Long[] ids, ModelMap modelMap) {
+        if (ids == null) {
+            logger.info("Delete incompleted: no comments selected");
+            return "/article";
+        } else {
+            Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+            String username = authentication.getName();
+            UserDTO userDTO = userService.getByUsername(username);
+            userDTO.setDeleted(false);
```

Comment:
No need to put it here, better to use default value

Reply:
done